### DISCUSSION TITLE
What is correct way to use same useFieldArray multiple times

### DISCUSSION THREAD

**Discussion by tparvi** [react-hook-form/react-hook-form#10141] [Q&A] [answered]
Codesandbox: https://codesandbox.io/s/rhf-use-field-array-twice-dqh44l

In the code I am using useFieldArray twice: once to render read only view and once to render edit view where you can add new person by clicking the button.

**Problem**: When I click _Add person_ only the edit view updates. I have added console.log to display the contents and you can see that `useWatch` returns the updated list but `useFieldArray` does not.

![image](https://user-images.githubusercontent.com/871053/226266667-f02552f8-aba7-4d07-a178-1971034e1a08.png)

**My understanding is that this is by design** because useFieldArray only contains the id of the item and some kind of link to the actual list. Sorry I don't know the technical details but I am interested in how this works.

**Question:** Is the correct way to implement this to use `setValue` or is there better way to handle this?

**Comment by Sajarin-M** [comment, marked as answer]
Hi @tparvi 👋
I don't know the technical details of `useFieldArray`,
but in this case, if you define fieldArray in your parent component and pass it as a prop to both `ReadOnlyList` and `EditList`, it will work as expected.
I hope someone else can clarify the technical details for you.
Please refer this [sandbox](https://codesandbox.io/s/usefieldarray-twice-ce7f7i?file=/src/App.tsx)

**Reply by BaileyMillerSSI** [reply]
I think this is one way to solve this but I would like to hear from the author if this behavior is expected? It seems like each call to useFieldArray creates it's own connection to the array and doesn't update the other instances when changes occur.

**Reply by tyler-dot-earth** [reply]
Old thread, just sharing that I agree that this feels like unintuitive/undocumented behavior.

**Comment by carlos7ags** [comment]
I have the same question as @BaileyMillerSSI. There are scenarios where you cannot pass it as prop.
